src/internal/Esp32_i2s.c

```diff
@@ -534,7 +547,11 @@ esp_err_t i2sSetSampleRate(uint8_t bus_num, uint32_t rate, uint8_t bits)
 
     // due to conf2.lcd_tx_wrx2_en being set for p8, bit rate is doubled
     // adjust by using bck here
+#if !defined(CONFIG_IDF_TARGET_ESP32S2) && !defined(CONFIG_IDF_TARGET_ESP32C3) && !defined(CONFIG_IDF_TARGET_ESP32S3)   
     uint8_t bck = (bits == 8 || bits == 16) ? 24 : 12;
+#else
+    uint8_t bck = (bits == 8 || bits == 16) ? 6 : 12;
```

Comment:
6?  More magic numbers.  I don't like these, normally I use constants that describe what they are but 8/16/24 were bit sizes I thought before.  Can you put a comment here on what the 6 does?

Reply:
Yes, we were talking about these magic number because there were here before my changes. Will add a comment.